keyboards/nifty_numpad/nifty_numpad.c

```diff
@@ -0,0 +1,225 @@
+/* Copyright 2023 Acliad
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#ifdef RGB_IDLE_ENABLE
+typedef struct {
+    uint8_t mode;
+    uint8_t speed;
+    HSV     hsv;
+} rgb_matrix_state_t;
```

Comment:
should add some asserts to ensure the sizes are consistent/don't change. 

```suggestion
} rgb_matrix_state_t;

_Static_assert(sizeof(rgb_matrix_state_t) == 5, "Invalid size for rgb_matrix_state_t");

```